mllib/src/main/scala/org/apache/spark/mllib/clustering/BisectingKMeans.scala

```diff
@@ -339,10 +339,14 @@ private object BisectingKMeans extends Serializable {
     assignments.map { case (index, v) =>
       if (divisibleIndices.contains(index)) {
         val children = Seq(leftChildIndex(index), rightChildIndex(index))
-        val selected = children.minBy { child =>
-          KMeans.fastSquaredDistance(newClusterCenters(child), v)
+        if (children.length > 0) {
```

Comment:
I added a filter on whether new cluster centers contains the children.  Would you be able to send me a sample dataset so I can validate the fix? Thank you!